src/csprojReader.ts

```diff
@@ -0,0 +1,31 @@
+import Nameable from "./nameable";
+const xml2js = require("xml2js");
+
+export default class CsprojReader implements Nameable {
+    private readonly xml: string;
+    private readonly xmlParser: any;
+
+    /**
+     * Initializes a new instance for a .csproj
+     * file.
+     *
+     * @param fileContent - The .csproj file full content
+     */
+    constructor(fileContent: string) {
+        this.xml = fileContent;
+        this.xmlParser = new xml2js.Parser();
+    }
+
+    public getRootNamespace(): string | undefined {
+        let foundNamespace = undefined;
+        this.xmlParser.parseString(this.xml, (error: any, result: any) => {
+            if (result === undefined
+                || result.Project.PropertyGroup === undefined
+                || result.Project.PropertyGroup[0].RootNamespace === undefined) {
```

Comment:
The RootNamespace can be in any PropertyGroup.
There is no predefined order for ProjectGroups.

Reply:
Thank You! Probably push a fix today.

Reply:
I'd guess something like this will do the trick:
`        this.xmlParser.parseString(this.xml, (error: any, result: any) => {
            if (result === undefined
                || result.Project.PropertyGroup === undefined
                || !result.Project.PropertyGroup.length) {
                return undefined;
            }

            for (let i = 0; i < result.Project.PropertyGroup.length; i++) {
                if (result.Project.PropertyGroup[i].RootNamespace) {
                    foundNamespace = result.Project.PropertyGroup[i].RootNamespace[0];
                    break;
                }
            }
        });`

Reply:
Sorry for the delay i have a deadline tomorrow and this is why i didn't push a fix yet, sorry!

Reply:
Thank You for the code!
I replaced the `for` loop with `foreach` to hopefully make it more readable  (and added a testcase too).